contracts/test/RepaymentCalc.t.sol

```diff
@@ -1,78 +1,63 @@
+}
 
-            assertEq(total,         total_calc);
-            assertEq(principal, principal_calc);
-            assertEq(interest,   interest_calc);
+contract RepaymentCalcTest is DSTest {
 
-            sumTotal += total;
+    function test_calcType() external {
+        assertTrue(new RepaymentCalc().calcType() == uint8(10));
+    }
 
-            bob.makePayment(address(loan1));
+    function test_name() external {
+        assertTrue(new RepaymentCalc().name() == "INTEREST_ONLY");
+    }
 
-            if (loan1.paymentsRemaining() > 0) {
-                assertEq(total,        lastTotal);
-                assertEq(interest,  lastInterest);
-                assertEq(total,         interest);
-                assertEq(principal,            0);
-            } else {
-                assertEq(total,     principal + interest);
-                assertEq(principal,              loanAmt);
-                withinPrecision(totalPaid, sumTotal, 8);
-                assertEq(beforeBal - IERC20(USDC).balanceOf(address(bob)), sumTotal);  // Pays back all principal, plus interest
-            }
+    function assert_nextPayment(
+        RepaymentCalc repaymentCalc,
+        uint256 apr,
+        uint256 paymentIntervalSeconds,
+        uint256 paymentsRemaining,
+        uint256 principalOwed,
+        uint256 expectedTotal,
+        uint256 expectedPrincipalOwed,
+        uint256 expectedInterest
+    ) internal {
+        MockLoan loan = new MockLoan(apr, paymentIntervalSeconds, paymentsRemaining, principalOwed);
+        (uint256 actualTotal, uint256 actualPrincipalOwed, uint256 actualInterest) = repaymentCalc.getNextPayment(address(loan));
+        assertEq(actualTotal,         expectedTotal);
+        assertEq(actualPrincipalOwed, expectedPrincipalOwed);
+        assertEq(actualInterest,      expectedInterest);
+    }
 
-            lastTotal    = total;
-            lastInterest = interest;
-        }
+    function test_getNextPayment() external {
+        RepaymentCalc repaymentCalc = new RepaymentCalc();
+        assert_nextPayment(repaymentCalc, 1_000_000, 500, 2, 1_000_000, 1_585,     0,         1_585);
+        assert_nextPayment(repaymentCalc, 1_000_000, 500, 1, 1_000_000, 1_001_585, 1_000_000, 1_585);
+        assert_nextPayment(repaymentCalc, 0,         500, 2, 1_000_000, 0,         0,         0);
+        assert_nextPayment(repaymentCalc, 0,         500, 1, 1_000_000, 1_000_000, 1_000_000, 0);
+        assert_nextPayment(repaymentCalc, 1_000_000, 0,   2, 1_000_000, 0,         0,         0);
+        assert_nextPayment(repaymentCalc, 1_000_000, 0,   1, 1_000_000, 1_000_000, 1_000_000, 0);
+        assert_nextPayment(repaymentCalc, 1_000_000, 500, 2, 0,         0,         0,         0);
+        assert_nextPayment(repaymentCalc, 1_000_000, 500, 1, 0,         0,         0,         0);
```

Comment:
It actually works out to be the same answers, but `apr` is actually the second param, so it should be swapped with `paymentIntervalSeconds`